tests/generic_config_updater/test_kubernetes_config.py

```diff
@@ -0,0 +1,300 @@
+    '"KUBERNETES_MASTER": {\n'
+    '        "SERVER": {\n'
+    '            "disable": "false",\n'
+    '            "ip": "k8svip.ap.gbl"\n'
+    '        }\n'
+    '    }'
+]
+K8SFULLCONFIG2 = [
+    '"KUBERNETES_MASTER": {\n'
+    '        "SERVER": {\n'
+    '            "disable": "false",\n'
+    '            "ip": "k8svip2.ap.gbl"\n'
+    '        }\n'
+    '    }'
+]
+K8SEMPTYIPCONFIG = [
+    '"KUBERNETES_MASTER": {\n'
+    '        "SERVER": {\n'
+    '            "disable": "true",\n'
+    '            "ip": ""\n'
+    '        }\n'
+    '    }'
+]
+
+
+# K8S config patch
+K8SEMPTYTOHALFPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER",
+        "value": {"SERVER": {}}
+    }
+]
+K8SHALFTOWRONGIPPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "true"
+    },
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": ""
+    }
+]
+K8SHALFTOFULLPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "false"
+    },
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": "k8svip.ap.gbl"
+    }
+]
+K8SFULLTOHALFPATCH = [
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER/SERVER/ip"
+    },
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER/SERVER/disable"
+    }
+]
+K8SHALFTOEMPTYPATCH = [
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER"
+    }
+]
+K8SEMPTYTOFULLPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER",
+        "value": {"SERVER": {"disable": "false", "ip": "k8svip.ap.gbl"}}
+    }
+]
+K8SFULLTOFULLPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": "k8svip2.ap.gbl"
+    }
+]
+K8SFULLTOWRONGIPPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": ""
+    },
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "true"
+    }
+]
+K8SFULLTOEMPTYPATCH = [
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER"
+    }
+]
+K8SEMPTYTOWRONGIPPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER",
+        "value": {"SERVER": {"disable": "true", "ip": ""}}
+    }
+]
+K8SWRONGIPTOFULLPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": "k8svip.ap.gbl"
+    },
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "false"
+    }
+]
+
+
+EMPTYK8STABLE = ()
+FULLK8STABLE = ("k8svip.ap.gbl", "false")
+FULLK8STABLE2 = ("k8svip2.ap.gbl", "false")
+WRONGIPK8STABLE = ("", "true")
+
+
+# Succeed and fail flag
+SUCCEED = "SUCCEED"
+FAIL = "FAIL"
+
+
+test_data_1 = {
+    0: (K8SEMPTYTOHALFPATCH, K8SHALFCONFIG, EMPTYK8STABLE, SUCCEED),
```

Comment:
>0 [](http://example.com/codeflow?start=4&length=1)

Maintaining the index is not necessary. Could you remove?

Reply:
Thanks for this comment, fixed.